linkerd/app/src/dst/default_profile.rs

```diff
@@ -0,0 +1,45 @@
+use super::Rejected;
+use futures::prelude::*;
+use linkerd2_app_core::{profiles, svc, Error};
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tracing::debug;
+
+pub fn layer<S>() -> impl svc::Layer<S, Service = RecoverDefaultProfile<S>> + Clone {
+    svc::layer::mk(|inner| RecoverDefaultProfile { inner })
+}
+
+#[derive(Clone, Debug)]
+pub struct RecoverDefaultProfile<S> {
+    inner: S,
+}
+
+impl<T, S> tower::Service<T> for RecoverDefaultProfile<S>
+where
+    S: tower::Service<T, Response = Option<profiles::Receiver>>,
+    S::Error: Into<Error>,
+    S::Future: Send + 'static,
+{
+    type Response = Option<profiles::Receiver>;
+    type Error = Error;
+    type Future =
+        Pin<Box<dyn Future<Output = Result<Option<profiles::Receiver>, Error>> + Send + 'static>>;
```

Comment:
nit/TIOLI: the closure in the `or_else` doesn't capture anything, so i think this _could_ be 
```suggestion
    type Future = future::OrElse<
        S::Future,
        future::Ready<Result<Self::Output, Self::Error>>
        fn(S::Error) -> future::Ready<Result<Self::Output, Self::Error>>,
     >;
```
if we wanted to remove the box. in practice, i don't know if the box here matters at all, but since we're already using combinators with a non-capturing closure, i thought it was worth pointing out.